contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/AbstractKuduOutputOperator.java

```diff
@@ -0,0 +1,666 @@
+    for (ColumnSchema primaryKeyInfo : primaryKeyColumns) {
+      primaryKeyColumnNames.add(primaryKeyInfo.getName());
+    }
+    Map<String,String> columnNameOverrides = getOverridingColumnNameMap();
+    if (columnNameOverrides == null) {
+      columnNameOverrides = new HashMap(); // to avoid null checks further down the line
+    }
+    for ( Field aFieldDef : classFields) {
+      String currentFieldName = aFieldDef.getName().toLowerCase();
+      if (allKuduTableColumnNames.contains(currentFieldName)) {
+        extractGetterForColumn(normalizedColumns.get(currentFieldName),aFieldDef);
+      } else {
+        if (columnNameOverrides.containsKey(aFieldDef.getName())) {
+          extractGetterForColumn(normalizedColumns.get(columnNameOverrides.get(aFieldDef.getName()).toLowerCase()),
+              aFieldDef);
+        }
+        if (columnNameOverrides.containsKey(aFieldDef.getName().toLowerCase())) {
+          extractGetterForColumn(normalizedColumns.get(columnNameOverrides.get(aFieldDef.getName().toLowerCase())
+              .toLowerCase()),aFieldDef);
+        }
+      }
+    }
+  }
+
+  /***
+   * Used to build a getter for the given schema column from the POJO field definition
+   * @param columnSchema The Kudu column definition
+   * @param fieldDefinition The POJO field definition
+   */
+  private void extractGetterForColumn(ColumnSchema columnSchema, Field fieldDefinition)
+  {
+    Type columnType = columnSchema.getType();
+    Class pojoClass = getTuplePayloadClass();
+    Object getter = null;
+    switch ( columnType ) {
+      case BINARY:
+        getter = PojoUtils.createGetter(pojoClass, fieldDefinition.getName(), ByteBuffer.class);
+        break;
+      case STRING:
+        getter = PojoUtils.createGetter(pojoClass, fieldDefinition.getName(), String.class);
+        break;
+      case BOOL:
+        getter = PojoUtils.createGetterBoolean(pojoClass, fieldDefinition.getName());
+        break;
+      case DOUBLE:
+        getter = PojoUtils.createGetterDouble(pojoClass, fieldDefinition.getName());
+        break;
+      case FLOAT:
+        getter = PojoUtils.createGetterFloat(pojoClass, fieldDefinition.getName());
+        break;
+      case INT8:
+        getter = PojoUtils.createGetterByte(pojoClass, fieldDefinition.getName());
+        break;
+      case INT16:
+        getter = PojoUtils.createGetterShort(pojoClass, fieldDefinition.getName());
+        break;
+      case INT32:
+        getter = PojoUtils.createGetterInt(pojoClass, fieldDefinition.getName());
+        break;
+      case INT64:
+      case UNIXTIME_MICROS:
+        getter = PojoUtils.createGetterLong(pojoClass, fieldDefinition.getName());
+        break;
+      default:
+        break;
+    }
+    if (getter != null) {
+      kuduColumnBasedGetters.put(columnSchema.getName(),getter);
+    }
+  }
+
+
+  public static ApexKuduConnection.ApexKuduConnectionBuilder usingKuduConnectionBuilder()
```

Comment:
This will be used in all concrete implementations of the AbstractKuduOutputOperator. This is allowing for a builder pattern to define the kudu table name, connection parameters and behaviour. The newly added BaseKuduOutputOperator which extends this abstract class uses this pattern to instantiate the underlying Async client to the Kudu cluster.